packages/notebook-extension/src/index.ts

```diff
@@ -2173,9 +2173,18 @@ function getCurrent(
   shell: JupyterFrontEnd.IShell,
   args: ReadonlyPartialJSONObject
 ): NotebookPanel | null {
-  const widget = args[SemanticCommand.WIDGET]
-    ? tracker.find(panel => panel.id === args[SemanticCommand.WIDGET]) ?? null
-    : tracker.currentWidget;
+  let widget: NotebookPanel | null = null;
+
+  // Check for panelId in args (used by cell toolbars)
+  if (args['panelId']) {
+    widget = tracker.find(panel => panel.id === args['panelId']) ?? null;
```

Comment:
I think this needs an update here too, ideally by importing `CellBarExtension` and passing the `WIDGET_ID_ARG`:

```suggestion
  if (args[CellBarExtension.WIDGET_ID_ARG]) {
    widget = tracker.find(panel => panel.id === args[CellBarExtension.WIDGET_ID_ARG]) ?? null;
```